lib/HLSL/HLOperationLower.cpp

```diff
@@ -2387,7 +2388,7 @@ Value *TranslatePrintf(CallInst *CI, IntrinsicOp IOP, DXIL::OpCode opcode,
                        HLObjectOperationLowerHelper *pObjHelper,
                        bool &Translated) {
   Translated = false;
-  CI->getContext().emitError(CI, "use of undeclared identifier 'printf'");
+  dxilutil::EmitErrorOnInstruction(CI, "use of undeclared identifier 'printf'");
```

Comment:
It's pre-existing, but I feel that this message could be more helpful. It's not undeclared nor unrecognized, it's just not supported by DXC (yet)